js/classes/entities/midground/chad.js

```diff
@@ -233,19 +245,28 @@ class Chad {
 
 
         // Dash action
-        // When not CHAD is not on the ground and his dash is allowed and the user is trying to dash then enter conditional.
-        if (GAME.user.dashing && !this.isOnGround && this.canDash) {
+        // console.log("DASH TIMER: " + this.dashTimer);
+        if (this.groundDashTimer > 0 && this.isOnGround) {
+            this.groundDashTimer -= GAME.clockTick;
+        } else if (this.groundDashTimer <= 0) {
+            this.canGroundDash = true;
+        }
+                    
+        const canDash = ((this.canAirDash && !this.isOnGround) || 
+                        (this.canGroundDash && this.isOnGround));
+
+        if (GAME.user.dashing && canDash) {
             if (!this.isDashing) {
+                // we just started dashing
+                ASSET_MGR.playSFX(SFX.SWOOSH.path, SFX.SWOOSH.volume);
                 this.xDashAnchoredOrigin = this.pos.x;
-                this.isDashing = true
-                ASSET_MGR.playAudio(SFX.SWOOSH.path, SFX.SWOOSH.volume);
+                this.isDashing = true;
             }
 
             // release wind particles every 0.05 seconds
-            if (GAME.gameTime % 0.05 < 0.01) {
-                GAME.addEntity(new ParticleEffect(Vector.add(this.scaleBoundingBoxOffset(), 
-                    new Vector(this.pos.x + this.scaledSize.x / 2, this.pos.y + this.scaledSize.y / 2)),
-                    ParticleEffect.WIND));
+            if (GAME.gameTime % 0.05 < 0.01) { // we use `< 0.01` instead of `== 0` to avoid floating point errors
```

Comment:
Interesting! I am curious why this happened. Does this fix seem to be foolproof?